plugin-server/src/worker/vm/lazy.ts

```diff
@@ -26,47 +26,87 @@ export class LazyPluginVM {
     resolveInternalVm!: Promise<PluginConfigVMResponse | null>
     totalInitAttemptsCounter: number
     initRetryTimeout: NodeJS.Timeout | null
+    ready: boolean
+    vmResponseVariable: string | null
 
     constructor() {
         this.totalInitAttemptsCounter = 0
         this.initRetryTimeout = null
+        this.ready = false
+        this.vmResponseVariable = null
         this.initVm()
     }
 
     public async getExportEvents(): Promise<PluginConfigVMResponse['methods']['exportEvents'] | null> {
-        return (await this.resolveInternalVm)?.methods.exportEvents || null
+        const exportEvents = (await this.resolveInternalVm)?.methods.exportEvents || null
+        if (!this.ready && exportEvents) {
+            await this.setupPluginIfNeeded()
+        }
+        return exportEvents
```

Comment:
can't we do something in typescript to pull all of this code into a function that we can use for all the functions? Would be a lot nicer later if we want to tweak something + readability

Reply:
yeah I'd originally thought of this but didn't immediately find a way to do it in TS. I have now.